homeassistant/components/ezviz/alarm_control_panel.py

```diff
@@ -0,0 +1,162 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import timedelta
+import logging
+
+from pyezviz import PyEzvizError
+from pyezviz.constants import DefenseModeType
+
+from homeassistant.components.alarm_control_panel import (
+    AlarmControlPanelEntity,
+    AlarmControlPanelEntityDescription,
+    AlarmControlPanelEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_NIGHT,
+    STATE_ALARM_DISARMED,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import (
+    DATA_COORDINATOR,
+    DOMAIN,
+    MANUFACTURER,
+)
+from .coordinator import EzvizDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=60)
+PARALLEL_UPDATES = 0
+
+
+@dataclass
+class EzvizAlarmControlPanelEntityDescriptionMixin:
+    """Mixin values for EZVIZ Alarm control panel entities."""
+
+    ezviz_alarm_states: list
+
+
+@dataclass
+class EzvizAlarmControlPanelEntityDescription(
+    AlarmControlPanelEntityDescription, EzvizAlarmControlPanelEntityDescriptionMixin
+):
+    """Describe a EZVIZ Alarm control panel entity."""
+
+
+ALARM_TYPE = EzvizAlarmControlPanelEntityDescription(
+    key="ezviz_alarm",
+    translation_key="ezviz_alarm",
+    ezviz_alarm_states=[
+        None,
+        STATE_ALARM_DISARMED,
+        STATE_ALARM_ARMED_AWAY,
+        STATE_ALARM_ARMED_NIGHT,
+    ],
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Ezviz alarm control panel."""
+    coordinator: EzvizDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
+        DATA_COORDINATOR
+    ]
+
+    async_add_entities([EzvizAlarm(coordinator, entry.entry_id)])
+
+
+class EzvizAlarm(CoordinatorEntity, AlarmControlPanelEntity):
+    """Representation of a Ezviz alarm control panel."""
+
+    coordinator: EzvizDataUpdateCoordinator
+    entity_description: EzvizAlarmControlPanelEntityDescription
+    _attr_has_entity_name = True
+    _attr_name = None
+    _attr_supported_features = (
+        AlarmControlPanelEntityFeature.ARM_AWAY
+        | AlarmControlPanelEntityFeature.ARM_NIGHT
+    )
+    _attr_code_arm_required = False
+
+    def __init__(self, coordinator: EzvizDataUpdateCoordinator, entry_id) -> None:
+        """Initialize alarm control panel entity."""
+        super().__init__(coordinator)
+        self._attr_unique_id = f"{entry_id}_{ALARM_TYPE.key}"
+        self.entity_description = ALARM_TYPE
+        self._attr_device_info: DeviceInfo = {
+            "identifiers": {(DOMAIN, "EZVIZ Alarm")},
+            "name": "EZVIZ Alarm",
```

Comment:
Is there always one device active on one HA instance? Can there be more?

Reply:
There can only be one per EZVIZ account. The integration also only caters for 1x account.

Reply:
How often do people change system? I think when you move to another place with another ezviz instance, the only thing that you will have is that your removed device is reused again for the new setup. Not sure if having device identifiers static is something what we want, so I'll leave this open for others to review

Reply:
Don't make it harder to cater for multiple instances of the integration in the future. Use the unique id for the entry if no other unique identifier exist to use for the device.